inc/saitunnel.h

```diff
@@ -0,0 +1,444 @@
+/*
+* Copyright (c) 2014 Microsoft Open Technologies, Inc. 
+*   
+*    Licensed under the Apache License, Version 2.0 (the "License"); you may 
+*    not use this file except in compliance with the License. You may obtain 
+*    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*
+*    THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR 
+*    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT 
+*    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS 
+*    FOR A PARTICULAR PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
+*
+*    See the Apache Version 2.0 License for specific language governing 
+*    permissions and limitations under the License. 
+*
+*    Microsoft would like to thank the following companies for their review and
+*    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+*    Dell Products, L.P., Facebook, Inc
+*   
+* Module Name:
+*
+*    saitunnel.h
+*
+* Abstract:
+*
+*    This module defines SAI Tunnel API
+*
+*/
+
+#if !defined (__SAITUNNEL_H_)
+#define __SAITUNNEL_H_
+
+#include <saitypes.h>
+
+/** \defgroup SAITUNNEL SAI - Next hop specific API definitions.
+ *
+ *  \{
+ */
+
+/**
+ * @brief Enum defining tunnel map types.
+ */
+typedef enum _sai_tunnel_map_type_t
+{
+    /** TUNNEL Map overlay ECN to underlay ECN   */
+    SAI_TUNNEL_MAP_OECN_TO_UECN = 0x00000001,
+
+    /** TUNNEL Map underlay ECN to overlay ECN   */
+    SAI_TUNNEL_MAP_UECN_TO_OECN = 0x00000002,
```

Comment:
the decap ECN mapper is slightly more complex than currently defined, in RFC6040, Section 4.2, the mapping maps (inner ECN, outer ECN) into final ECN.